### PR TITLE
forbid unsafe

### PR DESCRIPTION
**Pull request by tamasblummer**
Please consider this. I do not think performance hit justifies unsafe language use.

### CODE REVIEW

File: src/ln/onion_utils.rs [hunk 1]
```diff
@@ -113,9 +112,7 @@ pub(super) fn build_onion_payloads(route: &Route, starting_htlc_offset: u32) ->
 	let mut cur_value_msat = 0u64;
 	let mut cur_cltv = starting_htlc_offset;
 	let mut last_short_channel_id = 0;
-	let mut res: Vec<msgs::OnionHopData> = Vec::with_capacity(route.hops.len());
-	internal_traits::test_no_dealloc::<msgs::OnionHopData>(None);
-	unsafe { res.set_len(route.hops.len()); }
+	let mut res: Vec<msgs::OnionHopData> = vec![msgs::OnionHopData::default(); route.hops.len()];
```
Comment (TheBlueMatt) [inline src/ln/onion_utils.rs]: Would prefer to just insert(0, ...) at each loop iteration to make things even cleaner.

Comment (tamasblummer) [reply src/ln/onion_utils.rs]: do not know how to insert if not initialized. vec capacity != len

Comment (tamasblummer) [reply src/ln/onion_utils.rs]: if vec length is not 20*65 shift will fail

**Comment by TheBlueMatt** [comment]
Taken as #358.

File: src/ln/onion_utils.rs (hunk 1 above)
Comment (TheBlueMatt) [reply src/ln/onion_utils.rs]: Oops, I missed this, point was we don't shift on this vec, only on a later version of it, this one is always route.hops.len() in len at the end.

**Pull request closed without merging**